File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_hash = {number: 1 for number in phone_book}
    
    for phone_number in phone_book:
        temp = ""
        for digit in phone_number:
            temp += digit
            
            if temp in phone_hash and temp != phone_number:
                return False
                
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    const phoneHash = phone_book.reduce((acc, phone) => {
        acc[phone] = 1;
        return acc;
    }, {});
    
    for (let phone of phone_book) {
        let temp = "";
        for (let digit of phone) {
            temp += digit;
            
            if (temp in phoneHash && temp !== phone) {
                return false;
            }
        }
    }
    
    return true;
}